lighthouse-core/lib/page-functions.js

```diff
@@ -218,6 +218,40 @@ function getNodeSelector(node) {
   return parts.join(' > ');
 }
 
+/**
+ * @param {Element} node
+ * @returns {string}
+ */
+/* istanbul ignore next */
+function getNodeTitle(node) {
+  if (node.tagName === 'HTML' || node.tagName === 'BODY') {
+    // too broad, contains all page content
+    return null;
+  }
+  const title = node.innerText || node.getAttribute('alt') || node.getAttribute('aria-label');
+  if (title) {
+    return truncate(title, 80);
+  } else {
+    const nodeToUseForTitle = node.querySelector('[alt], [aria-label]');
+    if (nodeToUseForTitle) {
+      return getNodeTitle(nodeToUseForTitle);
+    }
+  }
+}
+
+/**
+ * @param {string} str
+ * @param {number} maxLength
+ * @returns {string}
+ */
+/* istanbul ignore next */
+function truncate(str, maxLength) {
+  if (str.length <= maxLength) {
+    return str;
+  }
+  return str.slice(0, maxLength - 1) + '…';
```

Comment:
could you add some tests for this edge case behavior?

`truncate('123', 3) === '123'`
`truncate('1234', 3) === '123…'`

or w/e way you expect truncate to work.